includes/templates/class-amp-post-template.php

```diff
@@ -260,11 +166,14 @@ private function build_post_content() {
 
 		$this->add_data_by_key( 'post_amp_content', $amp_content->get_amp_content() );
 		$this->merge_data_for_key( 'amp_component_scripts', $amp_content->get_amp_scripts() );
+		// Keeping both style arrays for backward compatibility
+		// post_amp_styles will be deprecated in the future
+		$this->merge_data_for_key( 'amp_styles', $amp_content->get_amp_styles() );
```

Comment:
I think to handle backwards compatibility here, we need to add a special handler in the `get` method that returns `amp_styles` when someone requests `post_amp_styles`. Maybe something like:

```
public function get( $property, $default = null ) {
    if ( 'post_amp_styles' === $property ) {
        _doing_it_wrong( __METHOD__, 'The `post_amp_styles` key has been deprecated; use `amp_styles` instead.', '0.6' );
        return $this->get( 'amp_styles', $default );
    } elseif ( 'amp_styles' === $property ) {
        // merge the values of  `$this->data[ 'post_amp_styles' ]` and `$this->data[ 'amp_styles' ]` and return
    }

    // ...
}
```